tests/include/drivers/keygen.h

```diff
@@ -29,11 +29,25 @@
 #if defined(PSA_CRYPTO_DRIVER_TEST)
 #include <psa/crypto_driver_common.h>
 
-extern void *test_driver_keygen_forced_output;
-extern size_t test_driver_keygen_forced_output_length;
-
-extern psa_status_t test_transparent_keygen_status;
-extern unsigned long test_transparent_keygen_hit;
+typedef struct {
+    /* If non-null, on success, copy this to the output. */
+    void *forced_output;
+    size_t forced_output_length;
+    /* If not PSA_SUCCESS, return this error code instead of processing the
+     * function call. */
+    psa_status_t forced_status;
+    /* Count the amount of times one of the keygen driver functions is called. */
+    unsigned long hits;
+} test_driver_keygen_hooks_t;
+
+#define TEST_DRIVER_KEYGEN_INIT { NULL, 0, PSA_ERROR_NOT_SUPPORTED, 0 }
```

Comment:
If you add a cast to the macro definition:
#define TEST_DRIVER_KEYGEN_INIT ( (test_driver_keygen_hooks_t){ NULL, 0, PSA_ERROR_NOT_SUPPORTED, 0 } ), you don't need the inline function. I think it would be better to do it that way as *_init() functions in the code base are used slightly differently.

Reply:
This `init` function parallel to the `INIT` macro follows the PSA model. Going forward, it's likely that we'll move to this model in Mbed TLS. A function is better because it's typed. The only advantage of the macro is that you can use it as a static analyzer, whereas a function can't be used there.

Reply:
ok, fine by me then.